Source/Utils/Inc/Utils/Math/Vector3.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#pragma once
+
+// -----------------------------------------------------------------------------
+
+namespace Utils
+{
+
+// -----------------------------------------------------------------------------
+
+struct Vector3
+{
+public:
+
+	Vector3();
+	Vector3( double X, double Y, double Z );
+
+	Vector3( const Vector3& Other ) = default;
+	Vector3( Vector3&& Other ) = default;
+
+	Vector3& operator=( const Vector3& Other ) = default;
+	Vector3& operator=( Vector3&& Other ) = default;
+
+	bool operator==( const Vector3& Other ) const;
+	bool operator!=( const Vector3& Other ) const;
+
+	double Norm() const;
+	double NormSquared() const;
+
+	double Dot( const Vector3& RHS ) const;
+	Vector3 Cross( const Vector3& RHS ) const;
+
+	Vector3 operator +( const Vector3& RHS ) const;
```

Comment:
Very minor but slight inconsistency with the spacing of operator overloads when compared to those on line 22. I think I've used the no space version elsewhere `Vector3 operator+( const Vector3& RHS ) const;`